Include/pystate.h

```diff
@@ -150,6 +162,8 @@ typedef struct _ts {
     PyObject *async_gen_firstiter;
     PyObject *async_gen_finalizer;
 
+    PyExcState *exc_info;
```

Comment:
Can we have `exc_info`, `exc_state` and `curexc_*` defined next to each other with comments explaining how we use each one of them?  Since you've already spent time to figure this stuff out anyways.